### PR TITLE
update signbit docs and add -0. to reference testing for unary and binary functions.

### LINKED ISSUE #53963: torch.signbit: incorrect output for negative zero
```python
>>> import torch
>>> import numpy as np
>>> t = torch.tensor(-0.)
>>> torch.signbit(t)
tensor(False)
>>> np.signbit(t.numpy())
True
```

C++ `std::signbit` works similar to NumPy.
https://en.cppreference.com/w/cpp/numeric/math/signbit

cc @ezyang @gchanan @zou3519 @bdhirsh @jbschlosser @anjali411 @mruberry @rgommers @heitorschueroff

### PR DESCRIPTION
**Pull request by kshitij12345**
Fixes https://github.com/pytorch/pytorch/issues/53963

### CODE REVIEW

**Comment by mruberry** [comment]
fyi https://github.com/pytorch/pytorch/pull/78216 happened -- but there may be parts of this PR beyond 78216 we want to base on top of it

also fyi @malfet who's looking at further improving signbit by short-circuiting on any unsigned type, not just bool

File: torch/_torch_docs.py [hunk 1]
```diff
@@ -2668,8 +2668,8 @@ def merge_dicts(*dicts):
 
 .. math::
     \text{out}_{i} = \begin{cases}
-        -|\text{input}_{i}| & \text{if} \text{other}_{i} \leq -0.0 \\
-        |\text{input}_{i}| & \text{if} \text{other}_{i} \geq 0.0 \\
+        -|\text{input}_{i}| & \text{if } \text{other}_{i} \leq -0.0 \\
+         |\text{input}_{i}| & \text{if } \text{other}_{i} \geq 0.0 \\
```
Comment (mruberry) [inline torch/_torch_docs.py:2672]: Nice formatting improvements

**Comment by mruberry** [comment]
Great stuff, @kshitij12345 -- I made a few inline comments for your review

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -2091,6 +2091,9 @@ def generate_elementwise_binary_small_value_tensors(
         math.pi + 0.00001,
     )
 
+    # Ref: https://github.com/pytorch/pytorch/issues/78413
+    _float_vals = (-0.,) if op.name != 'angle' else () + _float_vals
```
[the remark this answers is not in the document]
Comment (kshitij12345) [reply torch/testing/_internal/common_methods_invocations.py]: Updated.

**Comment by kshitij12345** [comment]
Gentle ping @mruberry

**Review by mruberry — APPROVED** [review]
Cool! Thanks @kshitij12345

**Pull request closed without merging**